stytch/consumer/api/m2m.py

```diff
@@ -27,3 +31,110 @@ def __init__(
+            data["scope"] = " ".join(scopes)
+
+        url = self.api_base.url_for(
+            "/v1/public/" + self.project_id + "/oauth2/token", data
+        )
+        res = self.sync_client.post(url, data)
+        return GetTokenResponse.from_json(res.response.status_code, res.json)
+
+    async def token_async(
+        self,
+        client_id: str,
+        client_secret: str,
+        scopes: Optional[List[str]] = None,
+    ) -> Optional[GetTokenResponse]:
+        """Retrieves an access token for the given M2M Client.
+        Access tokens are JWTs signed with the project's JWKs, and are valid for one hour after issuance.
+        M2M Access tokens contain a standard set of claims as well as any custom claims generated from templates.
+
+        Fields:
+          - client_id: The ID of the client.
+          - client_secret: The secret of the client.
+          - scopes: An array scopes requested. If omitted, all scopes assigned to the client will be returned.
+        """  # noqa
+
+        data: Dict[str, Any] = {
+            "grant_type": "client_credentials",
+            "client_id": client_id,
+            "client_secret": client_secret,
+        }
+        if scopes is not None:
+            data["scope"] = " ".join(scopes)
+
+        url = self.api_base.url_for(
+            "/v1/public/" + self.project_id + "/oauth2/token", data
+        )
+        res = self.async_client.post(url, data)
+        return GetTokenResponse.from_json(res.response.status_code, res.json)
+
+    # ENDMANUAL(m2m.token)
+
+    # MANUAL(m2m.authenticate_token)(SERVICE_METHOD)
+    # ADDIMPORT: from typing import Any, Dict, List, Optional
+    # ADDIMPORT: from stytch.consumer.models.m2m import AuthenticateTokenResponse
+    # ADDIMPORT: from stytch.consumer.api.sessions import Sessions
+    def authenticate_token(
+        self,
+        access_token: str,
+        required_scopes: Optional[List[str]] = None,
+        max_token_age: Optional[int] = None,
+    ) -> Optional[AuthenticateTokenResponse]:
+        """Validates a M2M JWT locally.
+        Note: There is no async version of this since we make no network calls.
+
+        Fields:
+          - access_token: The ID of the client.
+          - required_scopes: A list of scopes the token must have to be valid.
+          - max_token_age: The maximum possible lifetime in seconds for the token to be valid.
+        """  # noqa
+
+        sessions = Sessions(
+            self.api_base,
+            self.sync_client,
+            self.async_client,
+            self.jwks_client,
+            self.project_id,
+        )
+
+        resp = sessions.authenticate_m2m_jwt_local(
+            m2m_jwt=access_token, max_token_age_seconds=max_token_age
+        )
+        scopes = resp.scope.split(" ")
+        if required_scopes:
+            missing_scopes = filter(lambda scope: scope not in scopes, required_scopes)
+            if missing_scopes:
+                return None
```

Comment:
Use comprehensions instead of map/filter/reduce. Also, this can all be simplified.

One more nit-picky thing: I prefer to be really explicit with conditionals instead of relying on "truthy" values

```py
required_scopes = required_scopes or []
missing_scopes = [scope for scope in scopes if scope not in required_scopes]
if len(missing_scopes) != 0:
    return None
```